### ISSUE TITLE
garak adapter grades a NaN detector score as PASS

### ISSUE THREAD

**Issue by shaurya416** [Giskard-AI/giskard-oss#2840]
`libs/giskard-scan/src/giskard/scan/integrations/garak/_adapter.py:383-393`

```python
def _score_to_check(
    detector_label: str, score: float | None, detector: "Detector"
) -> CheckResult:
    """Map one detector score for one conversation onto a CheckResult."""
    details = _detector_details(detector_label)
    if score is None:
        return CheckResult.skip(message="detector returned no score", details=details)

    metrics = [Metric(name=detector_label, value=score)]
    if score > _HIT_THRESHOLD:
        return CheckResult.failure(
            message=detector.hit_desc, details=details, metrics=metrics
        )
    return CheckResult.success(
        message=detector.pass_desc, details=details, metrics=metrics
    )
```

Only `None` is treated as "the detector reached no verdict". Any other non-number falls through to the `>` comparison, and `nan > 0.5` is `False`, so a NaN score lands on the `success` branch. The comment at line 35 documents scores as being in `[0, 1]`, but nothing enforces that either: `-1.0` grades PASS and `2.0` / `inf` grade FAIL.

## Measured

The function body above was copied verbatim into a plain Python file (with `CheckResult` / `Metric` replaced by minimal stand-ins and a stub detector carrying `hit_desc` / `pass_desc`) and called directly:

| `score` | result | note |
|---|---|---|
| `1.0` | FAIL | control: confirmed hit |
| `0.0` | PASS | control: clean |
| `0.5` | PASS | boundary, documented as deliberate at line 35-37 |
| `None` | SKIP | control: "detector returned no score" |
| `float("nan")` | **PASS**, `Metric(value=nan)` | no verdict was reached |
| `float("inf")` | FAIL | outside documented range |
| `-1.0` | PASS | outside documented range |
| `2.0` | FAIL | outside documented range |

`Metric.value` is a plain `float` field (`core/result.py:123`), and pydantic v2 accepts NaN by default, so the NaN is carried into the result and into every export rather than rejected.

On reachability: garak's own detectors type `detect()` as `Iterable[float | None]` and none of the stock ones constructs a NaN on purpose. The realistic producers are an `HFDetector` whose pipeline emits NaN logits (the `(1.0 ± d["score"]) / 2` normalisation in `garak/detectors/base.py` passes it straight through) and any custom detector plugged into the adapter. The adapter already treats a different detector contract violation (more scores than conversations, lines 420-431) as something to surface rather than consume silently.

## Consequence

A conversation the detector could not score is reported as a clean pass with the detector's `pass_desc` message, and it counts in the PASS numerator of `SuiteResult.pass_rate`. #2759 wrote the convention "ERROR and SKIP are never coerced into pass or fail. Only an explicit verdict is." and its audit table (row 35) marked this adapter correct on the basis of the `None -> SKIP` branch; NaN was not in scope of that sweep. No test in `libs/giskard-scan/tests/integrations/garak/` exercises `_score_to_check` with a non-finite or out-of-range score (`_FakeDetector` in `test_garak_run.py` only feeds `0.9` and caller-chosen finite floats).

## Suggested fix

Guard with `math.isfinite` before the comparison, following the pattern already used in `libs/giskard-scan/src/giskard/scan/utils/knowledge_base.py:144,179-184` (which rejects non-finite embeddings with a comment explaining exactly this failure mode: a NaN comparison silently evaluates `False`). Something like:

```python
if score is None:
    return CheckResult.skip(message="detector returned no score", details=details)
if not math.isfinite(score) or not 0.0 <= score <= 1.0:
    return CheckResult.error(
        message=f"detector returned an invalid score {score!r} (expected a finite value in [0, 1])",
        details=details | {"score": repr(score)},
    )
```

ERROR rather than SKIP matches the deepteam adapter's reasoning at `integrations/deepteam/_adapter.py:177-183`: a value that is neither a verdict nor an honest "no score" is a detector fault, and ERROR is the only mapping no downstream rollup can turn into a pass. A test alongside `test_run_failure_score_marks_check_failed` feeding `float("nan")`, `-1.0` and `2.0` through `_FakeDetector` would pin it.

Happy to open the PR.
